pkg/bgpv1/manager/reconcile.go

```diff
@@ -420,29 +434,77 @@ func (r *LBServiceReconciler) lbServiceReconciler(
 		existingSelector = sc.Config.ServiceSelector
 	}
 
+	// Populate local availability used for externalTrafficPolicy=local handling
+	sla := make(svcLocalAvailability)
+endpointsLoop:
+	for _, eps := range r.epDiffStore.List() {
+		svc, exists, err := r.resolveSvcFromEndpoints(eps)
+		if err != nil {
+			// Cannot resolve service from endpoints. We have nothing to do here.
+			continue
+		}
+
+		if !exists {
+			// No service associated with this endpoint. We're not interested in this.
+			continue
+		}
+
+		// We only need Endpoints tracking for externalTrafficPolicy=Local
+		if svc.Spec.ExternalTrafficPolicy != slim_corev1.ServiceExternalTrafficPolicyTypeLocal {
+			continue
+		}
+
+		svcID := eps.ServiceID
+
+		for _, be := range eps.Backends {
+			if be.NodeName == cstate.CurrentNodeName {
+				// At least one endpoint is available on this node. We
+				// can make unavailable to available.
+				if available, found := sla[svcID]; !found || !available {
+					sla[svcID] = true
+				}
+				continue endpointsLoop
+			}
+		}
+
+		// Note that we may have multiple endpoints for one service when
+		// we use EndpointSlice. We shouldn't make available to unavailable.
+		if _, found := sla[svcID]; !found {
+			sla[svcID] = false
+		}
```

Comment:
`sla` map going to contain all services in the cluster which have ServiceExternalTrafficPolicyTypeLocal.

I wonder can we make map to contain services which have ServiceExternalTrafficPolicyTypeLocal and present locally. i.e. making `sla` map as a set of local services only. This reduces overall size of the map and also potentially simplifies the implementation.  

We can lookup in the map if svcID is present before iterating over eps.Backends, if present do not need to go over the backends. 

Pls correct me if you think my understanding is wrong.

Reply:
Yep, it sounds like a good idea. I fixed it.